src/components/AttachmentModal.tsx

```diff
@@ -1,186 +1,181 @@
-    source: PropTypes.oneOfType([PropTypes.string, sourcePropTypes]),
-
-    /** Optional callback to fire when we want to preview an image and approve it for use. */
-    onConfirm: PropTypes.func,
-
-    /** Whether the modal should be open by default */
-    defaultOpen: PropTypes.bool,
-
-    /** Optional callback to fire when we want to do something after modal show. */
-    onModalShow: PropTypes.func,
-
-    /** Optional callback to fire when we want to do something after modal hide. */
-    onModalHide: PropTypes.func,
-
-    /** Optional callback to fire when we want to do something after attachment carousel changes. */
-    onCarouselAttachmentChange: PropTypes.func,
-
-    /** Optional original filename when uploading */
-    originalFileName: PropTypes.string,
-
-    /** A function as a child to pass modal launching methods to */
-    children: PropTypes.func,
-
-    /** Whether source url requires authentication */
-    isAuthTokenRequired: PropTypes.bool,
-
-    /** Determines if download Button should be shown or not */
-    allowDownload: PropTypes.bool,
-
-    /** Title shown in the header of the modal */
-    headerTitle: PropTypes.string,
-
-    /** The report that has this attachment */
-    report: reportPropTypes,
-
-    /** The transaction associated with the receipt attachment, if any */
-    transaction: transactionPropTypes,
+type AttachmentModalOnyxProps = {
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
+    parentReport: OnyxEntry<OnyxTypes.Report>;
+    policy: OnyxEntry<OnyxTypes.Policy>;
+    parentReportActions: OnyxEntry<OnyxTypes.ReportActions>;
+    // eslint-disable-next-line react/no-unused-prop-types
+    session: OnyxEntry<OnyxTypes.Session>;
+};
 
-    ...withLocalizePropTypes,
+type Attachment = {
+    source: string;
+    isAuthTokenRequired: boolean;
+    file: File;
+    isReceipt: boolean;
+};
 
-    ...windowDimensionsPropTypes,
+type ImagePickerResponse = {
+    height: number;
+    name: string;
+    size: number;
+    type: string;
+    uri: string;
+    width: number;
+};
 
-    /** Denotes whether it is a workspace avatar or not */
-    isWorkspaceAvatar: PropTypes.bool,
+type Data = File | ImagePickerResponse;
 
-    /** Whether it is a receipt attachment or not */
-    isReceiptAttachment: PropTypes.bool,
+type ChildrenProps = {
+    displayFileInModal: (data: Data) => void;
+    show: () => void;
 };
 
-const defaultProps = {
-    source: '',
-    onConfirm: null,
-    defaultOpen: false,
-    originalFileName: '',
-    children: null,
-    isAuthTokenRequired: false,
-    allowDownload: false,
-    headerTitle: null,
-    report: {},
-    transaction: {},
-    onModalShow: () => {},
-    onModalHide: () => {},
-    onCarouselAttachmentChange: () => {},
-    isWorkspaceAvatar: false,
-    isReceiptAttachment: false,
+type AttachmentModalProps = AttachmentModalOnyxProps & {
+    source?: string;
```

Comment:
Why it's just a string? Please check what is passed as a `source` to this component, since before it was a more complex param. Maybe `AvatarSource` type will be more suitable

Reply:
yeah you are write , but I checked multiple places and everytime it was a string , maybe there is some specific place which I dont know? 
also there are places where the code is written like its expecting that it will be a string

Reply:
@kubabutkiewicz I just see the [examples](https://github.com/Expensify/App/blob/main/src/components/RoomHeaderAvatars.js#L34) where to source passed the result of `UserUtils.getFullSizeAvatar` function which has a return type `AvatarSource`.